codespell_lib/_codespell.py

```diff
@@ -502,8 +520,20 @@ def extract_words(text, word_regex, ignore_word_regex):
     return word_regex.findall(text)
 
 
+def apply_uri_ignore_words(check_words, line, word_regex, ignore_word_regex,
+                           uri_regex, uri_ignore_words):
+    if not uri_ignore_words:
+        return
+    for uri in re.findall(uri_regex, line):
+        for uri_word in extract_words(uri, word_regex,
+                                      ignore_word_regex):
+            if "*" in uri_ignore_words or uri_word in uri_ignore_words:
+                check_words.remove(uri_word)
```

Comment:
Can you not just drop the entire uri if `*` is present? Like your previous PR.

Also won't this currently mean that:
`I like to eat foo you can buy it from my online store at http://example.com/foo`

Would miss foo->food in the free text because it exists in the URI if I set `--uri-ignore-word-list=foo`?

Reply:
Edit, I see your test says not. Is it actually finding the right one, or just working because there are two hits on the line and it just removes one of them?

Reply:
It's working because there are two hits on the line, and removing one. codespell doesn't track where the hit was when doing spell corrections, so if there *is* a repeated misspelling on the line, and it's told to fix the spelling, it'll probably fix both. I don't think that's particularly bad though: while something might be ignored in a uri, if it's there in plain text, I'd expect the common case to be that it needs to either be fixed in both or ignored in both. (either way, this is an edge case scenario)

Reply:
I wonder if something even more like this might make more sense. Especially because your initial comment was people will just want to skip all typos in URIs wasn't it? Which means that 90% of the time we just want to hide the URI rather than process it twice.

```suggestion
    if "*" in uri_ignore_words:
      text = uri_regex.sub(' ', text)
    return word_regex.findall(text)


def apply_uri_ignore_words(check_words, line, word_regex, ignore_word_regex,
                           uri_regex, uri_ignore_words):
    for uri in re.findall(uri_regex, line):
        for uri_word in extract_words(uri, word_regex,
                                      ignore_word_regex):
            if uri_word in uri_ignore_words:
                check_words.remove(uri_word)
```

@larsoner do you have any strong feelings either way?

Reply:
No strong opinion, either way seems fine to me

Reply:
I've switched the code to check for "*". I don't think it's a performance issue, but whatever's easiest to understand.

I'm not clear what the suggested edit was intended for. If the top half was supposed to be rewriting extract_words, I'd note that extract_words is also used for the options.check_filenames code path, where I felt URI checks shouldn't be done (felt more likely to hit false positives).